### DISCUSSION TITLE
How to get CSRF token on the server

### DISCUSSION THREAD

**Discussion by swport** [nextauthjs/next-auth#8167] [Help]
So far I've seen this from the docs: https://next-auth.js.org/configuration/pages#credentials-sign-in
```typescript
import { getCsrfToken } from "next-auth/react"

///

export async function getServerSideProps(context: GetServerSidePropsContext) {
  return {
    props: {
      csrfToken: await getCsrfToken(context),  // isn't it client side function with a fetch call
    },
  }
}
```

Which I don't understand, because `getCsrfToken` is a client side function with a fetch call. Why do I want to make a fetch call to my own server from my own server?

**Comment by swport** [comment]
can someone help me make sense of this..???? I'm getting impatience of this stupidity.

**Reply by LuizCristino** [reply]
I think he is asking the same as this discussion https://github.com/nextauthjs/next-auth/discussions/7256
